qw11qD/platform.py

```diff
@@ -12,64 +15,72 @@
     """
     lo_map = {q: f"{q}/drive_lo" for q in ["D1", "D4", "D5"]}
     lo_map["D2"] = lo_map["D3"] = "D2D3/drive_lo"
 
     twpa_d = SGS100A(name="twpaD", address="192.168.0.33")
 
-    qubits = {}
-    for i in range(1, 6):
-        q = f"D{i}"
-        qubits[q] = Qubit(
-            name=q,
-            probe=IqChannel(
-                name=f"{q}/probe",
-                mixer=None,
-                lo=f"D/probe_lo",
-                acquisition=f"{q}/acquisition",
-            ),
-            acquisition=AcquireChannel(
-                name=f"{q}/acquisition", twpa_pump=twpa_d.name, probe=f"{q}/probe"
-            ),
-            drive=IqChannel(name=f"{q}/drive", mixer=None, lo=lo_map[q]),
-            flux=DcChannel(name=f"{q}/flux"),
+    qubits = {
+        f"D{i}": Qubit(
+            drive=f"D{i}/drive",
+            flux=f"D{i}/flux",
+            probe=f"D{i}/probe",
+            acquisition=f"D{i}/acquisition",
         )
+        for i in range(1, 6)
+    }
 
     # Connect logical channels to instrument channels (ports)
     # Readout
-    channels = [QmChannel(qubit.probe, "octave5", port=1) for qubit in qubits.values()]
+    channels = {
+        q.probe: IqChannel(device="octave5", path="1", mixer=None, lo="D/probe_lo")
+        for q in qubits.values()
+    }
     # Acquire
-    channels.extend(
-        QmChannel(qubit.acquisition, "octave5", port=1) for qubit in qubits.values()
-    )
+    channels |= {
+        q.acquisition: AcquireChannel(
+            device="octave5", path="1", twpa_pump=twpa_d.name, probe=q.probe
+        )
+        for q in qubits.values()
+    }
     # Drive
-    channels.extend(
-        [
-            QmChannel(qubits["D1"].drive, "octave5", port=2),
-            QmChannel(qubits["D2"].drive, "octave5", port=4),
-            QmChannel(qubits["D3"].drive, "octave5", port=5),
-            QmChannel(qubits["D4"].drive, "octave6", port=5),
-            QmChannel(qubits["D5"].drive, "octave6", port=3),
-        ]
-    )
+    channels |= {
+        qubits["D1"].drive: IqChannel(
+            device="octave5", path="2", mixer=None, lo=lo_map["D1"]
+        ),
+        qubits["D2"].drive: IqChannel(
+            device="octave5", path="4", mixer=None, lo=lo_map["D2"]
+        ),
+        qubits["D3"].drive: IqChannel(
+            device="octave5", path="5", mixer=None, lo=lo_map["D3"]
+        ),
+        qubits["D4"].drive: IqChannel(
+            device="octave6", path="5", mixer=None, lo=lo_map["D4"]
+        ),
+        qubits["D5"].drive: IqChannel(
+            device="octave6", path="3", mixer=None, lo=lo_map["D5"]
+        ),
+    }
     # Flux
-    channels.extend(
-        QmChannel(qubits[f"D{q}"].flux, "con9", port=q + 2) for q in range(1, 6)
-    )
+    channels |= {
+        qubits[f"D{q}"].flux: DcChannel(device="con9", path=str(q + 2))
+        for q in range(1, 6)
+    }
+    channels = cast(dict[ChannelId, Channel], channels)
```

Comment:
Is this really needed? I have an idea what this does (even though I had to search `typing.cast`), but it may look a bit weird to other users.

Reply:
I guess that to most users, using the VS Code Python extension, it doesn't look that weird :P

In any case, it was mainly for me, since I was using the type checker to refactor (here we don't have tests, other than actually loading and using the platform), so I reduced the noise with some extra hints.

Maybe I can simply declare the type during the original channel definition (as an empty dict).

Reply:
Replaced in c97da13